src/common/http.ts

```diff
@@ -77,7 +77,11 @@ export class TurboHTTPService implements TurboHTTPServiceInterface {
     });
 
     if (!allowedStatuses.includes(status)) {
-      throw new FailedRequestError(status, statusText);
+      throw new FailedRequestError(
+        status,
+        // Return error message from server if available
+        typeof data === 'string' ? data : statusText,
+      );
```

Comment:
typically payment/upload will respond with a detailed error message string, we can forward that info to integrations for better debugging